src/Psalm/Internal/Analyzer/Statements/Expression/ClassConstAnalyzer.php

```diff
@@ -844,6 +844,7 @@ private static function getOverriddenConstant(
                     assert($parent_classlike_storage !== null);
                     if (!isset($parent_classlike_storage->parent_interfaces[strtolower($interface)])
                         && !isset($interface_storage->parent_interfaces[strtolower($parent_classlike_storage->name)])
+                        && $interface_const_storage->stmt_location !== $parent_const_storage->stmt_location
```

Comment:
$interface_const_storage !== $parent_const_storage should be good too :)

Reply:
Done in e8bc9a620